core/src/main/scala/org/apache/spark/shuffle/ShuffleMemoryManager.scala

```diff
@@ -152,6 +199,11 @@ class ShuffleMemoryManager protected (
     taskMemory.remove(taskAttemptId).foreach { numBytes =>
       memoryManager.releaseExecutionMemory(numBytes)
     }
+    taskConsumers.remove(taskAttemptId).map { consumers =>
+      consumers.foreach { case (consumer, numBytes) =>
+        logWarning(s"Memory leak in task $taskAttemptId: $consumer $numBytes bytes")
```

Comment:
Same here; I think this should be an exception in unit tests and a warning in production.